### PR TITLE
my take at the ui.rs cleanup

### PR DESCRIPTION
**Pull request by Pebor**
Saw the comment about about it not being clean nor pretty, so here is my proposal.
That one error message is kinda weird :shrug:

### CODE REVIEW

File: src/ui.rs [hunk 1]
```diff
@@ -87,12 +87,12 @@ fn create_components(left: &Box, centered: &Box, right: &Box) {
...
-            .split(ALIGNMENT)
-            .next()
-            .expect("[ERROR] Invalid widget alignment!\n")
-            .to_uppercase();
+        // Grabs widget alignment and widget type from the identifier separated by '-̈́'.
+        let (widget_alignment, widget_type) = identifier.split_once(ALIGNMENT)
+            .expect("[ERROR] Widget should be named as [alignment]-[widget_type]_[name]");
+
+        // Formats the widget alignment.
+        let widget_alignment = widget_alignment.to_uppercase();
```
Comment (vars1ty) [inline src/ui.rs:95]: Could you perhaps rename this formatted variable to like, `f_widget_alignment` (f = formatted) so it's more explicit for people looking at the code?

File: src/ui.rs [hunk 2]
```diff
@@ -87,12 +87,12 @@ fn create_components(left: &Box, centered: &Box, right: &Box) {
         // Identifier example: `left-label_ABC` <= `left-label` is the IDENTIFIER, `ABC` is the NAME.
         let identifier = identifiers[0];
 
-        // Grabs the widget alignment.
-        let widget_alignment = key
-            .split(ALIGNMENT)
-            .next()
-            .expect("[ERROR] Invalid widget alignment!\n")
-            .to_uppercase();
+        // Grabs widget alignment and widget type from the identifier separated by '-̈́'.
```
Comment (vars1ty) [inline src/ui.rs:90]: `-̈́` ? You may wanna put in `-`, seems like your keyboard had a small seizure xd

File: src/ui.rs [hunk 3]
```diff
@@ -114,52 +114,58 @@ fn create_components(left: &Box, centered: &Box, right: &Box) {
...
+                if !has_started_cava {
+                    // Ensure it only calls update_bars once.
+                    cava::update_bars();
+                    has_started_cava = true;
+                }
+
+                cava.add(widget_name, alignment, left, centered, right)
+            },
+            _ => {
+                // You are stupid.
```
Comment (vars1ty) [inline src/ui.rs:166]: This comment can be removed, forgot to remove it several versions ago

**Comment by vars1ty** [comment]
Check the comments and when they are resolved, it's ready for merge as the rest LGTM.

File: src/ui.rs (hunk 2 above)
Comment (Pebor) [reply src/ui.rs:90]: huh, weird, alacritty doesn't even render it differently, no idea how I did it lol
image

**Comment by Pebor** [comment]
Should be resolved now.

**Comment by vars1ty** [comment]
Alright nice and thanks!

**Pull request merged**